badger_utils/token_utils/distribute_from_whales_realtime.py

```diff
@@ -53,7 +49,35 @@ def distribute_from_whales_realtime(
         tokens = TARGET_TOKENS
     # Normal Transfers
     for token_addr in tokens:
-        distribute_from_whale_realtime(recipient, percentage=percentage, token_address=token_addr)
+        token = interface.IERC20(token_addr)
+        target_whale = _get_whale(token_addr)
+        if recipient.balance() < 2 * 10 ** 18:
+            distribute_test_ether(recipient, Wei("5 ether"))
+        _top_up_whale_with_funds(recipient, target_whale)
+
+        token.transfer(
+            recipient, token.balanceOf(target_whale) * percentage, {"from": target_whale}
+        )
+        # This is needed because Ethplrorer API will raise exc if requests are made too often
+        # for the free API key
+        sleep(0.5)
+
+
+def distribute_from_whales_realtime_amount(
+        recipient: Account,
+        amount: int,
+        tokens: Optional[List[str]] = None) -> None:
+    if not tokens:
+        tokens = TARGET_TOKENS
+    for token_addr in tokens:
+        token = interface.IERC20(token_addr)
+        target_whale = _get_whale(token_addr)
+        if recipient.balance() < 2 * 10 ** 18:
+            distribute_test_ether(recipient, Wei("5 ether"))
+        if token.balanceOf(target_whale) < amount:
+            raise NotEnoughBalance(f"Whale {target_whale} has not enough balance")
+        _top_up_whale_with_funds(recipient, target_whale)
```

Comment:
missing an indent here i think

Reply:
No, why?

Reply:
maybe im not understanding fully, but i thought the idea was to (1) find a whale for given token, and (2) transfer some of that token to our test wallet. why do we also have to top up the whale..?

Reply:
ah i see you removed L79